streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -266,7 +268,7 @@ void retryWithBackoff(final Map<TaskId, Set<TopicPartition>> tasksToBeCreated, f
                     } catch (final LockException e) {
                         // ignore and retry
                         if (!retryingTasks.contains(taskId)) {
-                            log.warn("{} Could not create task {}. Will retry: ", logPrefix, taskId, e);
+                            log.warn("{} Could not create task {} due to {}. Will retry: ", logPrefix, taskId, e);
```

Comment:
Here is the context for this change (sorry I did not include it in the top description):

While trouble shooting some production issue by looking at the log4j entries, I found for WARN entries printing the stacktrace is a bit overkill as it 1) panic users who thought it was a severe issue (if it is indeed we should use ERROR); so I'm trying to only print the error message and exception name for WARN and only print stack trace for ERROR.

In fact as you can see from previous commits of this PR, I was intended to do that for both ERROR and WARN. But after some thoughts I feel it is appropriate to only update for WARN.

WDYT? @mjsax @dguy